### PR TITLE
Production HA config for kube-prometheus-stack Helm chart

### PR DESCRIPTION
**Pull request by desenhar**
https://trello.com/c/DDZYYAl7/950-production-ha-config-for-kube-prometheus-stack-helm-chart

Kube-prometheus-stack is currently deployed in its default configuration, with the exception of Ingress being enabled. Some config in our Terraform helm_release should be changed to support highly-available production deployments.

Changes that have been made, 
Changed replica count for Prometheus and Alertmanager to 3 for staging and production.
Added self monitor for kube-state-metrics.
Added persistant storage for Prometheus at 50Gi/gp3 and 10Gi/gp3 for Alertmanager
Changed Grafana timezone to Europe/London
Added PodAntiAffinity for Prometheus and Alertmanager, with topologyspreadconstraints.

### CODE REVIEW

**Comment by desenhar** [comment]
`alertmanager-kube-prometheus-stack-alertmanager-db-alertmanager-kube-prometheus-stack-alertmanager-0   Bound    pvc-43e88371-0ce5-43fd-bd38-8f488fce327c   10Gi       RWO            ebs-sc         22h
alertmanager-kube-prometheus-stack-alertmanager-db-alertmanager-kube-prometheus-stack-alertmanager-1   Bound    pvc-f09649d3-31a3-450d-9840-0ec32d005794   10Gi       RWO            ebs-sc         22h
alertmanager-kube-prometheus-stack-alertmanager-db-alertmanager-kube-prometheus-stack-alertmanager-2   Bound    pvc-9fee3ae0-3edc-4d97-9556-e3d642c1992f   10Gi       RWO            ebs-sc         22h
prometheus-kube-prometheus-stack-prometheus-db-prometheus-kube-prometheus-stack-prometheus-0           Bound    pvc-907a343b-09e5-4051-9e9b-b85ac79bf430   50Gi       RWO            ebs-sc         23h
prometheus-kube-prometheus-stack-prometheus-db-prometheus-kube-prometheus-stack-prometheus-1           Bound    pvc-ff5f618a-a7fb-4aec-bfc2-161f681eefe7   50Gi       RWO            ebs-sc         23h
prometheus-kube-prometheus-stack-prometheus-db-prometheus-kube-prometheus-stack-prometheus-2           Bound    pvc-5af31bb4-509f-4c24-8764-6e6b11d6496b   50Gi       RWO            ebs-sc         23h`

PVC showing bound storageclass with correct sizing.

File: terraform/deployments/cluster-services/kube_prometheus_stack.tf [hunk 1]
```diff
@@ -236,6 +237,35 @@ resource "helm_release" "kube_prometheus_stack" {
         }
       }]
     }
+    alertmanager = {
+      alertmanagerSpec = {
+        podAntiAffinity = "hard"
+        replicas        = var.kube_prometheus_stack_replica_count
+        topologySpreadConstraints = [{
```
Comment (fredericfran-gds) [inline terraform/deployments/cluster-services/kube_prometheus_stack.tf:244]: I think the pod antiaffinity and spreadcontraints are not needed, by default the EKS schedule spread the pods across the AZ. I introduced a new variable too (see here": https://github.com/alphagov/govuk-infrastructure/blob/c34054dd9ca45aff71fa19d5fb75f4fed345a4a7/terraform/deployments/cluster-services/variables.tf#L46) that you can set your replicas to.

File: terraform/deployments/cluster-services/kube_prometheus_stack.tf [hunk 2]
```diff
@@ -258,6 +288,36 @@ resource "helm_release" "kube_prometheus_stack" {
         }
         podMonitorSelectorNilUsesHelmValues     = false
         serviceMonitorSelectorNilUsesHelmValues = false
+        replicas                                = var.kube_prometheus_stack_replica_count
```
Comment (fredericfran-gds) [inline terraform/deployments/cluster-services/kube_prometheus_stack.tf:291]: use the new variable mentioned above

File: terraform/deployments/cluster-services/kube_prometheus_stack.tf [hunk 3]
```diff
@@ -258,6 +288,36 @@ resource "helm_release" "kube_prometheus_stack" {
...
+        podAntiAffinity                         = "hard"
+        topologySpreadConstraints = [{
```
Comment (fredericfran-gds) [inline terraform/deployments/cluster-services/kube_prometheus_stack.tf:293]: same comment as for alertmanager

**Review by fredericfran-gds — COMMENTED** [review]
have a look at my comments

File: terraform/deployments/variables/integration/common.tfvars [hunk 4]
```diff
@@ -41,3 +41,4 @@ rds_skip_final_snapshot = true
 secrets_recovery_window_in_days = 0
 
 default_desired_ha_replicas = 2
+
```
Comment (fredericfran-gds) [inline terraform/deployments/variables/integration/common.tfvars:44]: remove empty line added

File: terraform/deployments/variables/production/common.tfvars [hunk 5]
```diff
@@ -32,3 +32,5 @@ www_dns_validation_rdata = "fnvjfn8tfff6n003cf.fastly-validations.com"
 
 frontend_memcached_node_type   = "cache.t4g.medium"
 shared_redis_cluster_node_type = "cache.t4g.medium"
+
+kube_prometheus_stack_replica_count = 3
```
Comment (fredericfran-gds) [inline terraform/deployments/variables/production/common.tfvars:36]: no longer used, remove

**Review by fredericfran-gds — COMMENTED** [review]
looks good, thanks for making the changes, some minor cleanup required.

**Pull request merged**